instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/CoreRerouteTest.kt

```diff
@@ -281,19 +273,195 @@ class CoreRerouteTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::class.jav
+            longitude = originLocation.longitude()
+        }
+        mockLocationReplayerRule.loopUpdate(originLocationUpdate, times = 1)
+        val offRouteLocation = mockRouteReroute.routeWaypoints.first()
+        val offRouteLocationUpdate = mockLocationUpdatesRule.generateLocationUpdate {
+            latitude = offRouteLocation.latitude()
+            longitude = offRouteLocation.longitude()
+        }
+
+        mockWebServerRule.requestHandlers.addAll(mockRoute.mockRequestHandlers)
+        val requestHandler = MockDirectionsRequestHandler(
+            profile = DirectionsCriteria.PROFILE_DRIVING_TRAFFIC,
+            jsonResponse = mockRoute.routeResponseJson,
+            expectedCoordinates = mockRoute.routeWaypoints,
+            relaxedExpectedCoordinates = false
+        )
+        val rerouteRequestHandler = MockDirectionsRequestHandler(
+            profile = DirectionsCriteria.PROFILE_DRIVING_TRAFFIC,
+            jsonResponse = mockRouteReroute.routeResponseJson,
+            expectedCoordinates = listOf(
+                offRouteLocation,
+                mockRoute.routeWaypoints.last()
+            ),
+            relaxedExpectedCoordinates = true
+        )
+        val responseModifier = DelayedResponseModifier(4000) { it }
+        rerouteRequestHandler.jsonResponseModifier = responseModifier
+        mockWebServerRule.requestHandlers.add(requestHandler)
+        mockWebServerRule.requestHandlers.add(rerouteRequestHandler)
+
+        val rerouteStateTransitionAssertion = RerouteStateTransitionAssertion(
+            mapboxNavigation.getRerouteController()!!
+        ) {
+            requiredState(RerouteState.Idle)
+            requiredState(RerouteState.FetchingRoute)
+            requiredState(RerouteState.RouteFetched(RouterOrigin.Offboard))
+            requiredState(RerouteState.Idle)
+        }
+
+        val originalRoutes = mapboxNavigation.requestRoutes(
+            RouteOptions.builder()
+                .applyDefaultNavigationOptions()
+                .applyLanguageAndVoiceUnitOptions(activity)
+                .baseUrl(mockWebServerRule.baseUrl)
+                .coordinatesList(mockRoute.routeWaypoints)
+                .alternatives(true)
+                .build()
+        ).getSuccessfulResultOrThrowException().routes
+        mapboxHistoryTestRule.historyRecorder.startRecording()
+        mockLocationReplayerRule.loopUpdate(originLocationUpdate, times = 120)
+        mapboxNavigation.startTripSession()
+        mapboxNavigation.setNavigationRoutesAndWaitForUpdate(listOf(originalRoutes.first()))
+
+        mapboxNavigation.routeProgressUpdates().filter {
+            it.currentState == RouteProgressState.TRACKING
+        }.first()
+        mockLocationReplayerRule.stopAndClearEvents()
+        mockLocationReplayerRule.loopUpdate(offRouteLocationUpdate, times = 120)
+        // wait for OFF_ROUTE
+        mapboxNavigation.offRouteUpdates().filter { it }.first()
+        delay(2000)
```

Comment:
Why?
I hate delays because they may introduce flakiness. Although I understand that sometimes they are the only solution. If it's the case here, can you add a comment stating why it's required?

Reply:
I wanted to set the alternatives async to reflect the exact issue I'm testing against, and which is much more likely in real world. Alternatives wouldn't be set synchronously at the same time as off-route happened.

Reply:
Oh, so this delay guards nothing, it's just imitating real world? Phew, that's a relief. Could you add a comment stating that?